doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1352 @@
+nexthop = prefix,              ; IP addresses separated ',' (empty indicates no gateway). May indicate VXLAN endpoint if vni_label is non zero.
+intf = ifindex? PORT_TABLE.key ; zero or more separated by ',' (zero indicates no interface)
+vni_label = VRF.vni            ; New Field. zero or more separated by ',' (empty value for non-vxlan next-hops). May carry MPLS label in future.
+router_mac = mac_address       ; New Field. zero or more remote router MAC address separated by ',' (empty value for non-vxlan next-hops)
+blackhole = BIT                ; Set to 1 if this route is a blackhole (or null0)
+
+```
+
+**NEIGH_SUPPRESS_APP_VLAN_TABLE**
+
+Producer: VlanMgr
+
+Consumer: VlanOrch
+
+Description: New table for per VLAN neighbor suppression configuration.
+
+Schema:
+
+```
+; New table
+;Stores Neigh Suppress configuration per VLAN
+
+key             = SUPPRESS_VLAN_NEIGH_TABLE:"Vlan"vlanid		; VLAN with prefix "NEIGH_SUPPRESS"
+suppress_neigh  = "ON" / "OFF" ; Default "OFF"
+```
+
+### 4.2.3 STATE_DB changes
+
+**STATE_VXLAN_TUNNEL_TABLE**
+
+Producer: vxlanorch
+
+Consumer: show scripts
+
+Description: 
+
+- Holds the created tunnels - static or dynamic. 
+- There is a VXLAN Tunnel entry for every tunnel created. 
+- The SIP and DIP are added for tunnels. 
+- Tunnel Creation source is added. 
+
+Schema:
+
+```
+; New table
+; Defines schema for VXLAN State
+key             = VXLAN_TUNNEL:name    ; tunnel name
+; field         = value
+SRC_IP          = ipv4                 ; tunnel sip
+DST_IP          = ipv4                 ; tunnel dip 
+tnl_src         = "CLI"/"EVPN"  
+operstatus      = "oper_up"/"oper_down"  
+```
+
+### 4.2.4 COUNTER_DB changes
+
+Producer: syncd
+
+Consumer: show scripts
+
+Description:
+
+- Per Tunnel Rx/Tx packets and octet counters are added.
+- Existing counters table is used to add the tunnel counters.
+
+Schema:
+
+```
+; Existing table
+key                           = COUNTERS:tunnel_vid   ; tunnel oid
+; field                       = value
+SAI_TUNNEL_STAT_IN_OCTETS     = number    ;uint64 
+SAI_TUNNEL_STAT_IN_PACKETS    = number    ;uint32  
+SAI_TUNNEL_STAT_OUT_OCTETS    = number    ;uint64 
+SAI_TUNNEL_STAT_OUT_PACKETS   = number    ;uint32
+```
+
+
+
+## 4.3 Modules Design and Flows
+
+#### EVPN Control Plane and configurations
+
+FRR as EVPN control plane is assumed in this document. However, any BGP EVPN implementation can be used as control plane as long as it complies with the EVPN design proposed in this document.
+
+Following configurations are common for enabling EVPN L2/L3 services:
+
+- Configure VXLAN tunnel with SIP only representing the VTEP.
+- Configure EVPN NVO instance.
+- Configure BGP EVPN address-family and activate IPv4/IPv6 BGP neighbors.
+
+
+The corresponding CONFIG_DB entries are as follows. 
+
+
+```
+VXLAN_TUNNEL_TABLE|{{source_vtep_name}}
+    "src_ip" : {{ipv4_address}}
+    
+VXLAN_EVPN_NVO_TABLE|{{nvo_name}}
+    "source_vtep" : {{source_vtep_name}}
+```
```

Comment:
These 2 tables' are different with the sonic-swss-common schema.h [link](https://github.com/Azure/sonic-swss-common/pull/339)